dev_v2.md

```diff
@@ -0,0 +1,200 @@
+# Developer Setup Guide
+
+Welcome to the **Montage Project**! This guide will help you set up the project for local development.
+
+---
+
+## Overview
+
+The **Montage Project** is a web application with two main components:
+
+1. **Frontend**:
+   - Built with **Vue 3**.
+   - Includes **Wikimedia Codex** for UI components, **Axios** for API requests, and **Pinia** for state management.
+   - Compiled using **Vite** for fast builds.
+
+2. **Backend**:
+   - Built with **Clastic**, a Python framework.
+   - Uses various Python libraries such as:
+     - **SQLAlchemy**: Database interactions.
+     - **mwoauth**: Used for authentication with MediaWiki’s OAuth.
+     - **pymysql**: MySQL driver.
+   - Serves the frontend and exposes API endpoints for application functionality.
+
+---
+
+## Prerequisites
+
+Ensure the following are installed:
+- **Docker** and **Docker Compose**: [Install Docker](https://www.docker.com/products/docker-desktop).
+- **Node.js** (v16 or above): [Install Node.js](https://nodejs.org).
+- **Make**: Available on most Unix-based systems.
+
+---
+
+## Setting Up the Project
+
+### 1. Clone the Repository
+```bash
+git clone git@github.com:hatnote/montage.git
+cd montage
+```
+
+### 2. Start the Frontend Build Watcher
+```bash
+cd frontend
+npm install
+npm run watch:build
+```
+
+⚠️ **Frontend does not run in dev mode**: This will:watch for changes by using `chokidar` command and automatically copied to the backend's `static` and `template` directories. So there will not be any hot reloading. Developers must refresh the browser to see updates.
+
+Apart from `npm watch:build`, these are other commands:
+* `npm run lint`: Lint the code
+* `npm run format`: Format the code
+
+### 3. Use the Makefile to start the backend
+* Open a new termial tab and change directory to root of repo
+* Copy and edit `config.dev.yaml` based on `config.default.yaml`
+* (Optional) In `config.dev.yaml` there is a line for `dev_local_cookie_value`. To get it,
+log in to the local app in your browser, and then copy the value from the
+`clastic_cookie` in the apps' cookies. This is your login cookie.
+* (Optional) Add your username as the `superuser` in the config. (This will allow you to
+add `su_to=<any user>` to the backend, if you want to test submitting as another
+juror.)
```

Comment:
this is fine to mention, but I believe su_to is mostly used by automated tests these days :)